### PR TITLE
Spi internalized

### PR DESCRIPTION
**Pull request by Scooletz**
This PR internalizes API that was made public.

Requires rebase after #256 is merged

### CODE REVIEW

File: Hazelcast.Net/Hazelcast.Client.Spi/Exception.cs [hunk 1]
```diff
@@ -41,7 +40,7 @@ public RetryableHazelcastException(string message) : base(message)
     /// A <see cref="RetryableHazelcastException"/> that indicates operation is sent to a machine that isn't member of the cluster
     /// </summary>
     [Serializable]
-    public class TargetNotMemberException : RetryableHazelcastException
+    class TargetNotMemberException : RetryableHazelcastException
```
Comment (asimarslan) [inline Hazelcast.Net/Hazelcast.Client.Spi/Exception.cs]: This exception is defined in the protocol list so we need to keep it public.

File: Hazelcast.Net/Hazelcast.Client.Spi/Exception.cs [hunk 2]
```diff
@@ -54,7 +53,7 @@ public TargetNotMemberException(string message) : base(message)
     /// whether the invocation runs or not is indeterminate
     /// </summary>
     [Serializable]
-    public class TargetDisconnectedException : RetryableHazelcastException
+    class TargetDisconnectedException : RetryableHazelcastException
```
Comment (asimarslan) [inline Hazelcast.Net/Hazelcast.Client.Spi/Exception.cs]: This exception is defined in the protocol list so we need to keep it public.

**Comment by asimarslan** [comment]
I think we can remove all SPI naming too.

**Comment by zpqrtbnk** [comment]
We are about to release version 4.0 of the Hazelcast.Net client, and the code has been vastly refactored, rendering this PR... unfortunately irrelevant.

**Pull request closed without merging**